bot/check-result.sh

```diff
@@ -0,0 +1,483 @@
+
+success_msg="no message matching <code>${GP_req_missing}</code>"
+failure_msg="found message matching <code>${GP_req_missing}</code>"
+CoDeList=${CoDeList}$(add_detail ${MISSING} 0 "${success_msg}" "${failure_msg}")
+
+success_msg="found message(s) matching <code>${GP_no_missing}</code>"
+failure_msg="no message matching <code>${GP_no_missing}</code>"
+CoDeList=${CoDeList}$(add_detail ${NO_MISSING} 1 "${success_msg}" "${failure_msg}")
+
+success_msg="found message matching <code>${GP_tgz_created}</code>"
+failure_msg="no message matching <code>${GP_tgz_created}</code>"
+CoDeList=${CoDeList}$(add_detail ${TGZ} 1 "${success_msg}" "${failure_msg}")
+
+comment_details="${comment_details_fmt/__DETAILS_LIST__/${CoDeList}}"
+
+
+# first construct comment_artefacts_list, abbreviated CoArList
+# then use it to set comment_artefacts
+CoArList=""
+
+# TARBALL should only contain a single tarball
+if [[ ! -z ${TARBALL} ]]; then
+    # TODO add tarball details: size, num entries, modules, software pkgs, misc
+    # <dd>
+    #   <details>
+    #     <summary><code>eessi-2023.04-software-linux-x86_64-generic-1682696567.tar.gz</code></summary>
+    #     size: 234 MiB (245366784 bytes)<br/>
+    #     entries: 1234<br/>
+    #     modules under _2023.04/software/linux/x86_64/intel/cascadelake/modules/all/_<br/>
+    #     <pre>
+    #       GCC/9.3.0.lua<br/>
+    #       GCC/10.3.0.lua<br/>
+    #       OpenSSL/1.1.lua
+    #     </pre>
+    #     software under _2023.04/software/linux/x86_64/intel/cascadelake/software/_
+    #     <pre>
+    #       GCC/9.3.0/<br/>
+    #       CMake/3.20.1-GCCcore-10.3.0/<br/>
+    #       OpenMPI/4.1.1-GCC-10.3.0/
+    #     </pre>
+    #     other under _2023.04/software/linux/x86_64/intel/cascadelake/_
+    #     <pre>
+    #       .lmod/cache/spiderT.lua<br/>
+    #       .lmod/cache/spiderT.luac_5.1<br/>
+    #       .lmod/cache/timestamp
+    #     </pre>
+    #   </details>
+    # </dd>
+    size="$(stat --dereference --printf=%s ${TARBALL})"
+    size_mib=$((${size} >> 20))
+    tmpfile=$(mktemp --tmpdir=. tarfiles.XXXX)
+    tar tf ${TARBALL} > ${tmpfile}
+    entries=$(cat ${tmpfile} | wc -l)
+    # determine prefix from job config: VERSION/software/OS_TYPE/CPU_FAMILY/ARCHITECTURE
+    # 2023.04/software/linux/x86_64/intel/skylake_avx512
+    # repo_version = 2022.11
+    # software
+    # os_type = linux
+    # software_subdir = aarch64/generic
+    repo_version=$(cfg_get_value "repository" "repo_version")
+    os_type=$(cfg_get_value "architecture" "os_type")
+    software_subdir=$(cfg_get_value "architecture" "software_subdir")
+    prefix="${repo_version}/software/${os_type}/${software_subdir}"
+    modules_entries=$(grep "${prefix}/modules" ${tmpfile})
+    software_entries=$(grep "${prefix}/software" ${tmpfile})
+    lmod_entries=$(grep "${prefix}/.lmod/cache" ${tmpfile})
+    other_entries=$(cat ${tmpfile} | grep -v "${prefix}/modules" | grep -v "${prefix}/software")
```

Comment:
missing `grep -v` for `.lmod/cache`?

Reply:
Actually, I'd like to keep them under `other*`, so everything that is not under `software` and `modules` appears under `other`.

Reply:
makes sense